[Unittest or Build Fail] Launchpad/ARM

Case 1. 18.04 / armhf (CAPI unit test fail)
- https://launchpadlibrarian.net/437298176/buildlog_ubuntu-bionic-armhf.nnstreamer_0.2.1.0-0~201908141107~ubuntu18.04.1_BUILDING.txt.gz
- (note that 18.08 / arm64 is ok)

Case 2. 19.04 / armhf & arm64 (CAPI unit test fail)
- https://launchpadlibrarian.net/437298204/buildlog_ubuntu-disco-arm64.nnstreamer_0.2.1.0-0~201908141107~ubuntu19.04.1_BUILDING.txt.gz
- https://launchpadlibrarian.net/437298235/buildlog_ubuntu-disco-armhf.nnstreamer_0.2.1.0-0~201908141107~ubuntu19.04.1_BUILDING.txt.gz

- A few fails are from single-invoke timeout
- Another fail is from NOT-ANALYZED-YET.

Case 3. 19.10 / armhf & arm64 (BUILD FAIL / gcc9 becomes more strict!)
- https://launchpadlibrarian.net/437297993/buildlog_ubuntu-eoan-arm64.nnstreamer_0.2.1.0-0~201908141107~ubuntu19.10.1_BUILDING.txt.gz
- https://launchpadlibrarian.net/437297992/buildlog_ubuntu-eoan-armhf.nnstreamer_0.2.1.0-0~201908141107~ubuntu19.10.1_BUILDING.txt.gz
```
[123/145] Compiling C object 'tests/59830eb@@bmp2png@exe/bmp2png.c.o'.
FAILED: tests/59830eb@@bmp2png@exe/bmp2png.c.o 
cc -Itests/59830eb@@bmp2png@exe -Itests -I../tests -I/usr/include/libpng16 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Werror -std=gnu89 '-DVERSION="0.2.1"' -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Waggregate-return -Wold-style-definition -Wdeclaration-after-statement '-DNNSTREAMER_CONF_FILE="/etc/nnstreamer.ini"' -DHAVE_ORC=1 -DENABLE_TENSORFLOW_LITE=1 -DENABLE_PYTORCH=1 -DENABLE_CAFFE2=1 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -MD -MQ 'tests/59830eb@@bmp2png@exe/bmp2png.c.o' -MF 'tests/59830eb@@bmp2png@exe/bmp2png.c.o.d' -o 'tests/59830eb@@bmp2png@exe/bmp2png.c.o' -c ../tests/bmp2png.c
In file included from /usr/include/string.h:494,
                 from ../tests/bmp2png.c:36:
In function ‘strncpy’,
    inlined from ‘main’ at ../tests/bmp2png.c:291:3:
/usr/include/aarch64-linux-gnu/bits/string_fortified.h:106:10: error: ‘__builtin_strncpy’ specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../tests/bmp2png.c: In function ‘main’:
../tests/bmp2png.c:291:34: note: length computed here
  291 |   strncpy (pngfilename, argv[1], strlen (argv[1]) + 1);
      |                                  ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/tests/bmp2png.c
+++ b/tests/bmp2png.c
@@ -34,6 +34,7 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <string.h>
+#include <glib.h>
 
 typedef enum
 {
@@ -200,7 +201,6 @@ main (int argc, char *argv[])
   size_t size;
   char byte;
   char header[26];              /** gen24bBMP.py gives you 24B headered bmp file */
-  size_t len;
   int ret;
   char *pngfilename;
   int strn;
@@ -288,9 +288,7 @@ main (int argc, char *argv[])
   }
   fclose (bmpF);
 
-  len = strlen (argv[1]) + 1;
-  pngfilename = calloc (len, sizeof (char));
-  strncpy (pngfilename, argv[1], len);
+  pngfilename = g_strdup (argv[1]);
 
   /** Assume the last 4 characters are ".bmp" */
   strncpy (pngfilename + strlen (argv[1]) - 4, ".png", 5);

--- a/tests/meson.build
+++ b/tests/meson.build
@@ -18,7 +18,7 @@ libpng_dep = dependency('libpng', required: false)
 if libpng_dep.found()
   b2p = executable('bmp2png',
     'bmp2png.c',
-    dependencies: [libpng_dep],
+    dependencies: [libpng_dep, glib_dep],
     install: get_option('install-test'),
     install_dir: unittest_install_dir
   )

